lucene/core/src/java/org/apache/lucene/search/PointRangeQuery.java

```
@@ -369,6 +376,45 @@ public Scorer scorer(LeafReaderContext context) throws IOException {
         return scorerSupplier.get(Long.MAX_VALUE);
       }
 
+      @Override
+      public int count(LeafReaderContext context) throws IOException {
+        LeafReader reader = context.reader();
+
+        PointValues values = reader.getPointValues(field);
+        if (checkValidPointValues(values) == false) {
+          return 0;
+        }
+
+        if (reader.hasDeletions() == false
+            && numDims == 1
+            && values.getDocCount() == values.size()) {
+          // if all documents have at-most one point
+          final int[] intersectingLeafNodeCount = {0};
+          // create a custom IntersectVisitor that records the number of leafNodes that matched
+          final IntersectVisitor visitor =
+              new IntersectVisitor() {
+                @Override
+                public void visit(int docID) {
+                  intersectingLeafNodeCount[0]++;
```

Let's throw an UnsupportedOperationException here and move the increment to `visit(int,byte[])`? Tt would be a bug if this method would ever get called since the point is to skip nodes that are contained by the query.

---

I would really prefer all the count logic is in the method PointValues#pointCount. Let's consider the signature I propose in the issue. Then you just need to do the following:
```
return Math.toIntExact(values.pointCount(this::relate, this::matches));
```

---

By the way @jpountz, the method can be call because we call relate using the exact bounds of a leaf where visiting so we might end up calling this method :)

---

I'm good if we give this method a good signature.

---

One benefit of having it as a method of PointValues is that it can be better tested.

---

Done. Thanks for the idea @iverase. Looks much cleaner now (+ removes the inconsistency of adding the leaf node count separately).